app/src/main/java/de/danoeh/antennapod/activity/OnlineFeedViewActivity.java

```diff
@@ -416,14 +430,64 @@ private void showFeedInformation(final Feed feed, Map<String, String> alternateF
 
         viewBinding.backgroundImage.setColorFilter(new LightingColorFilter(0xff828282, 0x000000));
 
-        View header = View.inflate(this, R.layout.onlinefeedview_header, null);
-
-        viewBinding.listView.addHeaderView(header);
+        viewBinding.listView.addHeaderView(headerBinding.getRoot());
         viewBinding.listView.setSelector(android.R.color.transparent);
         viewBinding.listView.setAdapter(new FeedItemlistDescriptionAdapter(this, 0, feed.getItems()));
 
-        TextView description = header.findViewById(R.id.txtvDescription);
+        final int characterLimit = 160;
+        final boolean isLong = feed.getDescription().length() > characterLimit;
+
+        SpannableString originalDesc = null;
+        SpannableStringBuilder builder = new SpannableStringBuilder();
+
+        if (isLong) {
+            originalDesc = new SpannableString(HtmlToPlainText.getPlainText(feed.getDescription())
+                    .substring(0, characterLimit));
+            originalDesc.setSpan(new ForegroundColorSpan(
+                            headerBinding.txtvDescription.getTextColors().getDefaultColor()), 0,
+                    characterLimit, 0);
+
+            SpannableString readMore = new SpannableString(
+                    Html.fromHtml("&#160;" + "  <b>Read more</b>"));
```

Comment:
Why do you do some of the formatting with spans and some of it with html? This is not consistent

Reply:
Currently, depending on the screen size, it can happen that there is a line break between "read" and "more". That should be avoided. Either have both of them in a new line or both of them in the line above.